api/comms/signed_request.go

```diff
@@ -0,0 +1,49 @@
+package comms
+
+import (
+	"encoding/base64"
+	"errors"
+
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/gofiber/fiber/v2"
+)
+
+func ReadSignedPost(c *fiber.Ctx) ([]byte, string, error) {
+	if c.Method() != "POST" {
+		return nil, "", errors.New("readSignedPost bad method: " + c.Method())
+	}
+
+	payload := c.Body()
+
+	sigHex := c.Get(SigHeader)
+	wallet, err := recoverSigningWallet(sigHex, payload)
+	return payload, wallet, err
+}
+
+func recoverSigningWallet(signatureHex string, signedData []byte) (string, error) {
+	sig, err := base64.StdEncoding.DecodeString(signatureHex)
+	if err != nil {
+		err = errors.New("bad sig header: " + err.Error())
+		return "", err
+	}
+
+	// recover
+	hash := crypto.Keccak256Hash(signedData)
+	pubkey, err := crypto.SigToPub(hash[:], sig)
+	if err != nil {
+		return "", err
+	}
+
+	wallet := crypto.PubkeyToAddress(*pubkey).Hex()
+
+	// TODO: Still need this? We have a function for getting these in another file
```

Comment:
This is fine to remove. All we should need is this

recoverPubkeyFromCoreTx

which gets called from

https://github.com/AudiusProject/audius-protocol/blob/930bbe8a229322b13546b6023771d7334119179d/comms/discovery/pubkeystore/backfill.go#L21


We should probably do this in band rather than in a recovery process on a timer though. It would make things much more stable. cc @dharit-tan  too

Reply:
@raymondjacobson if I understand you correctly, we don't need to add keys to a store anymore, but we _do_ need to move the "recover from core" functionality over to this comms implementation?

Reply:
We probably do want to add them to the store, just not necessarily as a polling job. I think there are other consumers